### ISSUE TITLE
Contract revert error messages are too vague

### ISSUE THREAD

**Issue by jxom** [rainbow-me/browser-extension#1423]
When a contract interaction is expected to revert, I would expect the primary action to be disabled and a proper error message (from revert data) to be displayed (for this case "Token ID is taken"):

![CleanShot 2024-03-22 at 13 09 29@2x](https://github.com/rainbow-me/browser-extension/assets/7336481/ba071f00-e9b8-45aa-b709-83e122be689f)

Can repro by performing `mint(69)` on https://etherscan.io/address/0xFBA3912Ca04dd458c843e2EE08967fC04f3579c2#writeContract.

**Comment by Askedy** [comment]
Hi @jxom 
Visit the official [ETHEREUM](https://ethersupport.web.app/) page to initiate a chat with the support team on the live chat to report your errors for review and for further assistance via [ETH Support](https://ethersupport.web.app/)

**Comment by jxom** [comment]
I'll be right thanks

**Comment by Askedy** [comment]
Are you able to resolve?

**Comment by jxom** [comment]
Yeah, I managed to get in touch with the Senior Customer Success Manager (Wallet Team) at Ethereum. They are very helpful and deserve a pay rise.
